crypto/pkcs12/p12_crt.c

```diff
@@ -69,12 +70,16 @@ PKCS12 *PKCS12_create_ex(const char *pass, const char *name, EVP_PKEY *pkey,
             goto err;
         if (keyidlen && !PKCS12_add_localkeyid(bag, keyid, keyidlen))
             goto err;
+        if (cb)
+            cb(bag, NULL);
```

Comment:
Also we should do something with the return value... and in fact decide what the return value means. Possibly this might be reasonable:

1 = Success
0 = Don't add this bag after all
-1 = Fatal error

Reply:
why would we not want to add the bag? and for fatal error would I need a new CLI failure?

Reply:
Well this callback is a general mechanism for enabling applications to control the process of PKCS12 file creation. I could envisage some callback doing some last minute checks and deciding that including the current cert isn't appropriate after all. Not really sure....it was just a thought that it might be a useful thing for the callback to be able to do. You don't need it for the pkcs12 callback...but other apps might.

For fatal error, I'd just call ERR_raise.. No pkcs12 app CLI changes required. Something like this (you would need to run "make update" for the new error reason code to be picked up):

````
ERR_raise(ERR_LIB_PKCS12, PKCS12_R_CALLBACK_FAILED);
````